app/src/db/models/mixins.js

```diff
@@ -1,10 +1,29 @@
-function UpdatedAt (Model) {
+const moment = require('moment');
+
+/** Timestamps Objection Model Plugin
+ *  Add handlers to an Objection Model
+ *
+ *  In order to use JSON Schema Validation, we need to treat Timestamps as strings.
+ *  They still get stored as dates/timestamps, but in/out of the database they need to be strings.
+ *
+ *  This class will set the createdAt timestamp/string before insert and updatedAt before update.
+ *  The JSON Schema validation will pass as it goes through the marshalling, expecting createdAt and updateAt as strings.
+ *
+ * @see module:knex
+ * @see module:objection
+ */
+
+function Timestamps (Model) {
   return class extends Model {
-    $beforeUpdate (queryContext) {
-      super.$beforeUpdate(queryContext);
-      this.updatedAt = new Date();
+    async $beforeInsert(queryContext) {
+      await super.$beforeInsert(queryContext);
+      this.createdAt = moment().toISOString();
+    }
+    async $beforeUpdate(opt, queryContext) {
+      await super.$beforeUpdate(opt, queryContext);
+      this.updatedAt = moment().toISOString();
```

Comment:
Not completely sure if this needs to be async/awaited functions? Looks like the super functions need to complete before updating the respective field, but this could be wrapped as a Promise.resolve().then() chain instead to avoid needing to make this an explicit async function definition. Another option could be to update updatedAt and createdAt before invoking the super function, but not sure if that busts ordering.

Reply:
i believe you must call the super before any of your own code.